src/liballoc/collections/btree/search.rs

```diff
@@ -67,13 +67,12 @@ where
     Q: Ord,
     K: Borrow<Q>,
 {
-    // This function is defined over all borrow types (immutable, mutable, owned),
-    // and may be called on the shared root in each case.
+    // This function is defined over all borrow types (immutable, mutable, owned).
     // Using `keys()` is fine here even if BorrowType is mutable, as all we return
     // is an index -- not a reference.
     let len = node.len();
     if len > 0 {
```

Comment:
(too late now?) This test was explicitly added to exclude the shared root.

Reply:
It's not strictly too late if @Mark-Simulacrum wants to update that.

Reply:
Ah, yes, good point -- I'll try to fix this in a future PR. I think a more thorough review pass of which conditions and checks we still need would indeed be good.